specs/PrintToPdf.md

```diff
@@ -0,0 +1,325 @@
+# Background
+
+Developers have requested the ability to programmatically print to PDF in WebView2. Printing to PDF uses a virtual printer to create a PDF file from the current page. Currently the end user can do this from the print preview dialog by choosing the "Save as PDF" or "Microsoft Print to PDF" options. This API gives developers the ability to print to PDF without requiring the end user to go through the print preview dialog.
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+
+# Description
+This API consists of an asynchronous PrintToPdf method and a PrintSettings object. The PrintToPdf method accepts a path that the PDF file should be saved to.
+
+Use the CreateDefaultPrintSettings method to create a PrintSettings object with default values, which can be modified. The settings consist of: orientation, scale factor, page height and width, margins, printing of backgrounds, printing selection only, printing header and footer, header title, and footer URI. See API Details below for the default values.
+
+Currently other programmatic printing is not supported.
+
+# Examples
+```cpp
+// Shows the user a file selection dialog, then uses the selected path when
+// printing to PDF. If `useDefaultOrientation` is true, the page is printed
+// in portrait mode, otherwise the page is printed in landscape mode.
+void FileComponent::PrintToPdf(bool useDefaultOrientation)
+{
+    WCHAR defaultName[MAX_PATH] = L"WebView2_PrintedPdf.pdf";
+    OPENFILENAME openFileName = CreateOpenFileName(defaultName,
+                                                   L"PDF File\0*.pdf\0");
+    if (GetSaveFileName(&openFileName))
+    {
+        wil::com_ptr<ICoreWebView2Environment5> webviewEnvironment5;
+        CHECK_FAILURE(m_appWindow->GetWebViewEnvironment()->QueryInterface(
+            IID_PPV_ARGS(&webviewEnvironment5)));
+
+        wil::com_ptr<ICoreWebView2SPrintSettings> printSettings;
+        if (webViewEvironment5)
```

Comment:
Either add a comment to say why we don't need a nullptr check (because the new environment interface and corewebview2 interface come in the same version) or add an additional check.